packages/xpath/src/functions/xforms/geo.ts

```diff
@@ -114,3 +116,70 @@ export const distance = new NumberFunction(
 		return toAbsolutePrecision(sum(distances), PRECISION);
 	}
 );
+
+/**
+ * Returns whether a geopoint is inside the specified geoshape; aka 'geofencing'
+ * @param point the geopoint location to check for inclusion.
+ * @param polygon the closed list of geoshape coordinates defining the polygon 'fence'.
+ * @return true if the location is inside the polygon; false otherwise.
+ *
+ * Adapted from https://wrfranklin.org/Research/Short_Notes/pnpoly.html:
+ *
+ * int pnpoly(int nvert, float *vertx, float *verty, float testx, float testy) {
+ *     int i, j, c = 0;
+ *     for (i = 0, j = nvert - 1; i < nvert; j = i++) {
+ *         if (((verty[i] > testy) != (verty[j] > testy)) &&
+ *             (testx < (vertx[j] - vertx[i]) * (testy - verty[i]) / (verty[j] - verty[i]) + vertx[i]))
+ *             c = !c;
+ *     }
+ *     return c;
+ * }
+ */
+const calculateIsPointInGPSPolygon = (point: Geopoint, polygon: Geotrace) => {
```

Comment:
[In the specs](https://docs.getodk.org/form-operators-functions/#geofence) 

> Returns True if the specified point is inside the specified geoshape, False otherwise.

Is it safe to assume that it excludes points that are vertices or points on the segment (perimeter line)? 
I tested this, and it returns `false`.

Reply:
Answered below - it is undefined in the algorithm.